wom/src/main/scala/wdl4s/wdl/Scatter.scala

```diff
@@ -28,3 +24,34 @@ case class Scatter(index: Int, item: String, collection: WdlExpression, ast: Ast
 
   override def toString: String = s"[Scatter fqn=$fullyQualifiedName, item=$item, collection=${collection.toWdlString}]"
 }
+
+object Scatter {
+  val FQNIdentifier = "$scatter"
+
+  /**
+    * @param index Index of the scatter block. The index is computed during tree generation to reflect wdl scatter blocks structure.
+    */
+  def apply(ast: Ast, index: Int): Scatter = {
+    val item = ast.getAttribute("item").asInstanceOf[Terminal].getSourceString
+    new Scatter(index, item, WdlExpression(ast.getAttribute("collection")), ast)
+  }
+
+  def womScatterNode(scatter: Scatter, localLookup: Map[String, GraphNodePort.OutputPort]): ErrorOr[ScatterNodeWithInputs] = {
+    val scatterCollectionExpression = WdlWomExpression(scatter.collection, Option(scatter))
+    val scatterVariableSourceValidation = WdlWomExpression.findInputsforExpression(scatter.item, scatterCollectionExpression, localLookup, Map.empty)
+    val scatterItemTypeValidation = scatterCollectionExpression.evaluateType(localLookup.map { case (k, v) => k -> v.womType }) flatMap {
+      case WdlArrayType(itemType) => Valid(itemType)
+      case other => s"Cannot scatter over a non-array type ${other.toWdlString}".invalidNel
+    }
+
+    val innerGraphAndScatterItemInputValidation: ErrorOr[(Graph, RequiredGraphInputNode)] = for {
+      scatterItemType <- scatterItemTypeValidation
+      womInnerGraphScatterVariableInput = RequiredGraphInputNode(scatter.item, scatterItemType)
+      g <- WdlGraphNode.buildWomGraph(scatter, Set(womInnerGraphScatterVariableInput), localLookup)
+    } yield (g, womInnerGraphScatterVariableInput)
+
+    (scatterVariableSourceValidation, innerGraphAndScatterItemInputValidation).tupled flatMap { case (scatterVariableSource, (innerGraph, scatterItemInputNode)) =>
+      ScatterNode.scatterOverGraph(innerGraph, scatterVariableSource, scatterItemInputNode, localLookup) // + (scatter.item -> scatterItemInputNode.singleOutputPort))
```

Comment:
ToL: What's the comment at the end of this line? Is it an "also knows as" translation of the first half of the line? Or is it old code?